tao_compiler/mlir/disc/transforms/fusion_utils.cc

```diff
@@ -456,16 +621,25 @@ void FusionPattern::calculateOperandsAndResults() {
       assert(inserted);
 
       bool has_external_user = false;
+      bool has_internal_user = false;
       for (Operation* user : getValueUsers(v)) {
+        if (op == user) {
+          continue;
+        }
         if (!op_set.contains(user) && !inSameFusionFamily(user, op)) {
           has_external_user = true;
-          break;
+        }
+        if (op_set.contains(user)) {
+          has_internal_user = true;
         }
       }
 
       if (has_external_user) {
         results_.push_back(v);
         root_ops_.push_back(op);
+        if (!has_internal_user) {
```

Comment:
what's the difference between `results_` and `external_only_results_ `? Are they actual the same one?

Reply:
A result may be read by other ops inside the fusion, while an external only result are only read by ops outside the fusion.